spring-integration-core/src/main/java/org/springframework/integration/aggregator/AbstractCorrelatingMessageHandler.java

```diff
@@ -780,8 +781,10 @@ protected void afterRelease(MessageGroup group, Collection<Message<?>> completed
 
 	protected void forceComplete(MessageGroup group) { // NOSONAR Complexity
 		Object correlationKey = group.getGroupId();
+		Assert.notNull(correlationKey, "'correlationKey' must not be null");
 		// UUIDConverter is no-op if already converted
 		UUID groupId = UUIDConverter.getUUID(correlationKey);
+		Assert.notNull(groupId, "groupId is must not be null");
```

Comment:
It won't be null after `Assert.notNull(correlationKey)`.
Let's consider to mark those `UUIDConverter` methods with `@Contract("null -> null; !null -> !null")`.
See `StringUtils.quote()`, for example.